pkg/eks/create.go

```diff
@@ -463,3 +474,177 @@ func getParameterValueFromOutput(key string, outputs []*cloudformation.Output) s
 
 	return ""
 }
+
+// EnableEBSCSIDriverInput holds the options for enabling the EBS CSI driver
+type EnableEBSCSIDriverInput struct {
+	EKSService   services.EKSServiceInterface
+	IAMService   services.IAMServiceInterface
+	CFService    services.CloudFormationServiceInterface
+	Config       *eksv1.EKSClusterConfig
+	AddonVersion string
+}
+
+// EnableEBSCSIDriver manages the installation of the EBS CSI driver for EKS, including the
+// creation of the OIDC Provider, the IAM role and the validation and installation of the EKS add-on
+func EnableEBSCSIDriver(opts EnableEBSCSIDriverInput) error {
+	oidcID, err := configureOIDCProvider(opts.IAMService, opts.EKSService, opts.Config)
+	if err != nil {
+		return fmt.Errorf("could not configure oidc provider: %v", err)
+	}
+	roleArn, err := createEBSCSIDriverRole(opts.CFService, opts.Config, oidcID)
+	if err != nil {
+		return fmt.Errorf("could not create ebs csi driver role: %v", err)
+	}
+	_, err = installEBSAddon(opts.EKSService, opts.Config, roleArn, opts.AddonVersion)
+	if err != nil {
+		return fmt.Errorf("failed to install ebs csi driver addon: %v", err)
+	}
+
+	return nil
+}
+
+// CheckEBSAddon checks if the EBS CSI driver add-on is installed. If it is, it will return
+// the ARN of the add-on. If it is not, it will return an empty string. Otherwise, it will return an error
+func CheckEBSAddon(eksService services.EKSServiceInterface, config *eksv1.EKSClusterConfig) (string, error) {
+	input := eks.DescribeAddonInput{
+		AddonName:   aws.String(ebsCSIAddonName),
+		ClusterName: aws.String(config.Spec.DisplayName),
+	}
+
+	output, err := eksService.DescribeAddon(&input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
```

Comment:
you can use `errors.As` here https://pkg.go.dev/errors, it makes the code a bit more readable

Reply:
I was thinking about how could I implement this with `errors.As` but I don't know how to get rid of the `err.(awserr.Error)` to cast it to a `awserr` that I can then compare against `eks.ErrCodeResourceNotFoundException`, which is just an error code represented as string.

```go
var genericAWSErr awserr.Error
if errors.As(err, &genericAWSErr) {
	if aerr, _ := err.(awserr.Error); aerr.Code() == eks.ErrCodeResourceNotFoundException {
		// resource not found
	}
}
```

Is there a way I could make it easier and more readable?

Reply:
```golang
var genericAWSErr awserr.Error
if errors.As(err, &genericAWSErr) && aerr.Code() == eks.ErrCodeResourceNotFoundException {
}
```
can something like this work?